### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by natoscott**

### CODE REVIEW

**Comment by Explorer09** [comment]
I remember you argued about "non-null-pointer-assert" being just "line noise in the code".

**Comment by natoscott** [comment]
> 

Yes, that's correct.  That's not what this change is doing.

**Comment by Explorer09** [comment]
@natoscott Well, I mean while some assert lines are not useful by themselves, they might convey the intent of the programmers. This way I can see when the lack of null pointer check is intentional, and not a developer's oversight.

I know the proper way to mark the pointer as being non-null is to use `__attribute__((nonnull))`. The assert statement is an old way to mark the same intent. It's not just useless code noise and that's my point.

**Comment by natoscott** [comment]
> 

Sure, I understand your POV.  A comment can often be a better way to go in that situation though.

**Review by cgzones — APPROVED** [review]

File: linux/Platform.c [hunk 1]
```diff
@@ -291,12 +291,12 @@ void Platform_getLoadAverage(double* one, double* five, double* fifteen) {
    *fifteen = NAN;
 }
 
-int Platform_getMaxPid(void) {
+pid_t Platform_getMaxPid(void) {
+   int maxPid = 4194303;
```
[the remark this answers is not in the document]
Comment (natoscott) [reply linux/Platform.c]: Thanks @BenBE - fixed

File: unsupported/Platform.c [hunk 2]
```diff
@@ -103,7 +103,7 @@ void Platform_getLoadAverage(double* one, double* five, double* fifteen) {
 }
 
 int Platform_getMaxPid(void) {
-   return 1;
+   return INT_MAX;
 }
```
[the remark this answers is not in the document]
Comment (natoscott) [reply unsupported/Platform.c]: Fixed, thanks

**Pull request merged**